### PR TITLE
feat(api): optimize api compatibility

### LINKED ISSUE #5398: Optimization for certain API Interfaces
# Rationale

Currently, java-tron's HTTP interfaces support GET and POST requests. However, there are some inconsistencies in parameter validation and response between GET and POST calls for the same interface. Here are some examples:

### case1: incorrect log level for invalid parameter

`/wallet/getapprovedlist
`
When passing an invalid parameter like modifying `type_url` to `type_url1`, the server logs an ERROR:

    18:02:57.889 ERROR [qtp464658113-329] [capsule](TransactionCapsule.java:350) Type of the Any message does not match the given class.

### case2: inconsistent naming convention for parameters

`/wallet/getavailableunfreezecount
`
image

The GET and POST requests have inconsistent naming for the parameter `ownerAddress` (GET) vs `owner_address` (POST).

### case3: GET parameter handling contradicts business logic 

`/wallet/getcanwithdrawunfreezeamount
`
image

For the `timestamp` parameter, the GET request throws `NumberFormatException` if missing while POST handles it as 0 by default, which aligns with business logic.

`/wallet/getcandelegatedmaxsize
`
image

For the `type` parameter, GET request throws `NumberFormatException` if missing while POST handles it as 0 by default, which aligns with business logic.

`/wallet/getblockbynum
`
image
For the `num` parameter, POST resolves it to 0 without passing it, and GET throws an error. To make it compatible, GET should be consistent with POST.

### case4：Lack of compatibility handling for parameter

`/wallet/getaccount
`
image

The `visible` parameter for GET only works for base64 encoded addresses, lacking compatibility handling.

### case5: POST directly invokes GET

`/wallet/getnowblock`
`/wallet/listnodes`
`/wallet/getassetissuelist`
`/wallet/listproposals`
image

For these APIs, POST directly calls GET, so the `{visible\:true} `input is not handled properly.

# Implementation

To summarize, 5 cases are listed where GET and POST have inconsistencies. The solutions are as follows:

case1：change log level for invalid parameter validation

case2：standardize naming conventions

case3：align GET handling with POST logic

case4：add compatibility handling for GET parameters

case5：modify POST response based on input

### PR DESCRIPTION
**Pull request by lxcmyf**
**What does this PR do?**
optimize api compatibility, close #5398

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

**Comment by halibobo1205** [comment]
@lxcmyf Please fix CI , Add unit test to cover all changes.

File: framework/src/main/java/org/tron/core/services/http/ListProposalsServlet.java [hunk 1]
```diff
@@ -8,29 +9,43 @@
...
       Util.processError(e, response);
     }
   }
 
+  @Override
   protected void doPost(HttpServletRequest request, HttpServletResponse response) {
-    doGet(request, response);
+    try {
+      PostParams params = PostParams.getPostParams(request);
+      boolean visible = Util.getVisible(request);
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/core/services/http/ListProposalsServlet.java:33]: Why get it again here?

File: framework/src/main/java/org/tron/core/services/http/GetAccountServlet.java [hunk 2]
```diff
@@ -17,6 +17,7 @@ public class GetAccountServlet extends RateLimiterServlet {
   @Autowired
   private Wallet wallet;
 
+  @Override
```
Comment (halibobo1205) [inline framework/src/main/java/org/tron/core/services/http/GetAccountServlet.java:20]: No PR-related changes are recommended.

File: framework/src/main/java/org/tron/core/services/http/Util.java [hunk 3]
```diff
@@ -338,7 +340,7 @@ public static boolean getVisiblePost(final String input) {
     if (StringUtil.isNotBlank(input)) {
       JSONObject jsonObject = JSON.parseObject(input);
       if (jsonObject.containsKey(VISIBLE)) {
-        visible = jsonObject.getBoolean(VISIBLE);
+        visible = Boolean.parseBoolean(jsonObject.getString(VISIBLE));
```
[the remark this answers is not in the document]
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/Util.java:343]: This is meaningful, when the requested ContentType is in the application/x-www form urlencoded format, it is necessary to parse the true string format.

File: framework/src/main/java/org/tron/core/services/http/ListProposalsServlet.java (hunk 1 above)
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/ListProposalsServlet.java:33]: To be compatible with the post method for obtaining parameters, when the visible obtained in the URL is false, it is then obtained from the post method.

File: framework/src/main/java/org/tron/core/services/http/GetAssetIssueListServlet.java [hunk 4]
```diff
@@ -8,29 +9,43 @@
...
       Util.processError(e, response);
     }
   }
 
+  @Override
   protected void doPost(HttpServletRequest request, HttpServletResponse response) {
-    doGet(request, response);
+    try {
+      PostParams params = PostParams.getPostParams(request);
+      boolean visible = Util.getVisible(request);
```
[the remark this answers is not in the document]
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/GetAssetIssueListServlet.java:33]: Yes, if it is a post request, the obtained visible is true.

File: chainbase/src/main/java/org/tron/core/capsule/TransactionCapsule.java [hunk 5]
```diff
@@ -346,6 +346,10 @@ public static byte[] getOwner(Transaction.Contract contract) {
         }
       }
       return owner.toByteArray();
+    } catch (InvalidProtocolBufferException invalidProtocolBufferException) {
+      logger.warn("InvalidProtocolBufferException occurred because {}, please verify the interface "
+          + "input parameters", invalidProtocolBufferException.getMessage());
+      return new byte[0];
```
[the remark this answers is not in the document]
Comment (lxcmyf) [reply chainbase/src/main/java/org/tron/core/capsule/TransactionCapsule.java:352]: Throwing an exception violates compatibility.

File: framework/src/main/java/org/tron/core/services/http/PostParams.java [hunk 6]
```diff
@@ -21,6 +24,9 @@ public PostParams(String params, boolean visible) {
   public static PostParams getPostParams(HttpServletRequest request) throws Exception {
     String input = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
     Util.checkBodySize(input);
+    if (!validJson(input)) {
```
[the remark this answers is not in the document]
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/PostParams.java:27]: it is necessary.

File: framework/src/main/java/org/tron/core/services/http/Util.java [hunk 7]
```diff
@@ -569,4 +571,23 @@ public static void validateParameter(String contract) throws InvalidParameterExc
...
+  public static boolean validJson(String str) {
+    try {
+      JSON.parse(str);
+    } catch (Exception e) {
+      return false;
+    }
+    return true;
+  }
+
+  public static String getJsonString(String input) {
```
[the remark this answers is not in the document]
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/Util.java]: ok

File: framework/src/main/java/org/tron/core/services/http/GetAssetIssueListServlet.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (halibobo1205) [reply framework/src/main/java/org/tron/core/services/http/GetAssetIssueListServlet.java:33]: I suggest changing it to whether or not to include this parameter `visible`, if not , fetch from `PostParams`.

[the remark this answers is not in the document]
Comment (lxcmyf) [reply framework/src/main/java/org/tron/core/services/http/GetAssetIssueListServlet.java:33]: This is not impossible.

**Review by 317787106 — APPROVED** [review]

**Review by halibobo1205 — APPROVED** [review]
LGTM

**Pull request merged**